p2p/switch.go

```diff
@@ -491,11 +492,15 @@ func (sw *Switch) listenerRoutine(l Listener) {
 			break
 		}
 
-		// ignore connection if we already have enough
-		// leave room for MinNumOutboundPeers
-		maxPeers := sw.config.MaxNumPeers - DefaultMinNumOutboundPeers
-		if maxPeers <= sw.peers.Size() {
-			sw.Logger.Info("Ignoring inbound connection: already have enough peers", "address", inConn.RemoteAddr().String(), "numPeers", sw.peers.Size(), "max", maxPeers)
+		// Ignore connection if we already have enough peers.
+		_, in, _ := sw.NumPeers()
+		if in >= sw.config.MaxNumInboundPeers {
```

Comment:
note: if MaxNumInboundPeers is negative, number of conns is unlimited => which can cause consensus crash due to not enough file descriptors. Should we check and panic?

Reply:
We should probably catch issues in the config much sooner. Maybe we add ValidateBasic methods to the config struct we can call after parsing?

Reply:
I like this idea